### PR TITLE
test: add new dns checker pod

### PR DESCRIPTION
**Pull request by nebril**
related: #13430

### CODE REVIEW

**Comment by nebril** [comment]
test-missed-k8s

**Comment by nebril** [comment]
test-missed-k8s

**Comment by nebril** [comment]
test-missed-k8s

**Comment by nebril** [comment]
test-missed-k8s

**Comment by nebril** [comment]
test-missed-k8s

**Comment by nebril** [comment]
test-me-please

**Comment by nebril** [comment]
test-missed-k8s

File: test/helpers/kubectl.go [hunk 1]
```diff
@@ -2020,24 +2021,27 @@ func (kub *Kubectl) WaitForKubeDNSEntry(serviceName, serviceNamespace string) er
...
 
-		res := kub.ExecInFirstPod(ctx, LogGathererNamespace, logGathererSelector(false), fmt.Sprintf(digCMD, serviceNameWithNamespace, dnsClusterIP))
+		res := kub.ExecInFirstPod(ctx, LogGathererNamespace, DnsCheckSelector, fmt.Sprintf(digCMD, serviceNameWithNamespace, dnsClusterIP))
 		if res.err != nil {
 			logger.Debugf("failed to run dig in log-gatherer pod")
 			return false
 		}
-		serviceIPFromDNS := res.SingleOut()
-		if !govalidator.IsIP(serviceIPFromDNS) {
+		hasIPAddress, serviceIPFromDNS := hasIPAddress(res.Stdout())
```
Comment (christarazi) [inline test/helpers/kubectl.go]: I think we have a helper to split by lines, so I don't think you need to split the output in `hasIPAddress`/

**Comment by nebril** [comment]
test-missed-k8s

**Comment by nebril** [comment]
test-missed-k8s

**Comment by nebril** [comment]
test-missed-k8s

**Comment by nebril** [comment]
This ended up a non-ci specific issue, closing.

**Pull request closed without merging**